### PR TITLE
Add SCC Describer

### PR DESCRIPTION
**Pull request by pweil-**
fork PR: https://github.com/openshift/kubernetes/pull/5

Adds a describer for SCC.  Example output:

```
[pweil@localhost origin]$ oc describe scc anyuid
Name:				anyuid
Priority:			10
Access:				
  Users:			
  Groups:			system:cluster-admins
Settings:			
  Allow Privileged:		false
  Default Add Capabilities:	
  Required Drop Capabilities:	KILL,MKNOD,SYS_CHROOT,SETUID,SETGID
  Allowed Capabilities:		
  Allow Host Dir Volumes:	false
  Allow Host Network:		false
  Allow Host Ports:		false
  Allow Host PID:		false
  Allow Host IPC:		false
  Run As User Strategy:		
    Type:			RunAsAny
  SELinux Context Strategy:	
    Type:			MustRunAs
  FSGroup Strategy:		
    Type:			RunAsAny
    Ranges:			
  Supplemental Groups Strategy:	
    Type:			RunAsAny
    Ranges:			


[pweil@localhost origin]$ oc describe scc privileged
Name:				privileged
Access:				
  Users:			system:serviceaccount:openshift-infra:build-controller
  Groups:			system:cluster-admins,system:nodes
Settings:			
  Allow Privileged:		true
  Default Add Capabilities:	
  Required Drop Capabilities:	
  Allowed Capabilities:		
  Allow Host Dir Volumes:	true
  Allow Host Network:		true
  Allow Host Ports:		true
  Allow Host PID:		true
  Allow Host IPC:		true
  Run As User Strategy:		
    Type:			RunAsAny
  SELinux Context Strategy:	
    Type:			RunAsAny
  FSGroup Strategy:		
    Type:			RunAsAny
    Ranges:			
  Supplemental Groups Strategy:	
    Type:			RunAsAny
    Ranges:			
```

Fixes https://github.com/openshift/origin/issues/7051

### CODE REVIEW

**Comment by smarterclayton** [comment]
Looks good, please add some representative tests

**Comment by deads2k** [comment]
@stevekuznetsov ptal

**Comment by stevekuznetsov** [comment]
fixes https://github.com/openshift/origin/issues/5299

File: Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/describe.go [hunk 1]
```diff
@@ -72,18 +73,19 @@ func (e ErrNoDescriber) Error() string {
...
+		api.Kind("Service"):                    &ServiceDescriber{c},
+		api.Kind("ServiceAccount"):             &ServiceAccountDescriber{c},
+		api.Kind("Node"):                       &NodeDescriber{c},
+		api.Kind("LimitRange"):                 &LimitRangeDescriber{c},
+		api.Kind("ResourceQuota"):              &ResourceQuotaDescriber{c},
+		api.Kind("PersistentVolume"):           &PersistentVolumeDescriber{c},
+		api.Kind("PersistentVolumeClaim"):      &PersistentVolumeClaimDescriber{c},
+		api.Kind("Namespace"):                  &NamespaceDescriber{c},
+		api.Kind("Endpoints"):                  &EndpointsDescriber{c},
+		api.Kind("SecurityContextConstraints"): &SecurityContextConstraintsDescriber{c},
```
Comment (stevekuznetsov) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/describe.go]: Why are endpoint and scc kinds plural?

Comment (pweil-) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/describe.go]: that's how they've been for a while.  There is a special case for them in rest mapping for when the singular == the plural.

Comment (stevekuznetsov) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/describe.go]: /twitch

File: Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/describe.go [hunk 2]
```diff
@@ -413,6 +415,97 @@ func describeQuota(resourceQuota *api.ResourceQuota) (string, error) {
...
+		fmt.Fprintf(out, "  Required Drop Capabilities:\t%s\n", capsToString(scc.RequiredDropCapabilities))
+		fmt.Fprintf(out, "  Allowed Capabilities:\t%s\n", capsToString(scc.AllowedCapabilities))
+		fmt.Fprintf(out, "  Allow Host Dir Volumes:\t%t\n", scc.AllowHostDirVolumePlugin)
+		fmt.Fprintf(out, "  Allow Host Network:\t%t\n", scc.AllowHostNetwork)
+		fmt.Fprintf(out, "  Allow Host Ports:\t%t\n", scc.AllowHostPorts)
+		fmt.Fprintf(out, "  Allow Host PID:\t%t\n", scc.AllowHostPID)
+		fmt.Fprintf(out, "  Allow Host IPC:\t%t\n", scc.AllowHostIPC)
+
+		fmt.Fprintf(out, "  Run As User Strategy:\t\n")
+		fmt.Fprintf(out, "    Type:\t%s\n", string(scc.RunAsUser.Type))
```
Comment (stevekuznetsov) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/describe.go]: Most of these `Types:` statements should go after the previous line, e.g.
```go
fmt.Fprintf(out, "  Run As User Strategy:\t%s\n", string(scc.RunAsUser.Type))
```

**Comment by stevekuznetsov** [comment]
@pweil- need to update generated bash completions

File: Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/describe.go (hunk 1 above)
Comment (liggitt) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/describe.go]: > /twitch

yes. tough.

**Comment by pweil-** [comment]
@stevekuznetsov - updated - PTAL

**Comment by stevekuznetsov** [comment]
Could you post example output for the changes?

**Comment by pweil-** [comment]
Changes to move type inline
```
[pweil@localhost origin]$ oc describe scc restricted
Name:						restricted
Access:						
  Users:					
  Groups:					system:authenticated
Settings:					
  Allow Privileged:				false
  Default Add Capabilities:			
  Required Drop Capabilities:			KILL,MKNOD,SYS_CHROOT,SETUID,SETGID
  Allowed Capabilities:				
  Allow Host Dir Volumes:			false
  Allow Host Network:				false
  Allow Host Ports:				false
  Allow Host PID:				false
  Allow Host IPC:				false
  Run As User Strategy: MustRunAsRange		
  SELinux Context Strategy: MustRunAs		
  FSGroup Strategy: RunAsAny			
    Ranges:					
  Supplemental Groups Strategy: RunAsAny	
    Ranges:					
```

**Comment by pweil-** [comment]
actually, I'll me add a len check on the ranges as well.  Let me know if there is anything else you see

**Comment by smarterclayton** [comment]
Instead of empty, print "<none>" or "<empty>"

**Comment by stevekuznetsov** [comment]
I don't really understand why the formatting looks so awful if you're using a TabWriter, it looks good other than the empty representation. It would be nice if things lined up and didn't look so awful, but I can't tell why that's occurring.

**Comment by pweil-** [comment]
```
[pweil@localhost origin]$ oc describe scc restricted
Name:						restricted
Access:						
  Users:					<none>
  Groups:					system:authenticated
Settings:					
  Allow Privileged:				false
  Default Add Capabilities:			<none>
  Required Drop Capabilities:			KILL,MKNOD,SYS_CHROOT,SETUID,SETGID
  Allowed Capabilities:				<none>
  Allow Host Dir Volumes:			false
  Allow Host Network:				false
  Allow Host Ports:				false
  Allow Host PID:				false
  Allow Host IPC:				false
  Run As User Strategy: MustRunAsRange		
    UID:					<none>
    UID Range Min:				<none>
    UID Range Max:				<none>
  SELinux Context Strategy: MustRunAs		
    User:					<none>
    Role:					<none>
    Type:					<none>
    Level:					<none>
  FSGroup Strategy: RunAsAny			
    Ranges:					<none>
  Supplemental Groups Strategy: RunAsAny	
    Ranges:					<none>
```

**Comment by smarterclayton** [comment]
LGTM

**Comment by pweil-** [comment]
[merge]

**Pull request merged**